lib/fileHandler/index.js

```diff
@@ -104,9 +117,142 @@ module.exports = function (options) {
+			}
+
+			if (_.isUndefined(params.readStream) || _.isUndefined(params.readStream.pipe)) {
+				return reject(new Error('The object passed in must contain the property \'readStream\', referecing a read stream.'));
+			}
+
+			var client = knox.createClient({
+				key: options.aws.key,
+				secret: options.aws.secret,
+				bucket: 'workflow-file-uploads',
+				region: 'us-west-2'
+			});
+
+
+			// Generate a random file name
+			var fileName = guid();
+
+			var mimeType = params.contentType || mime.lookup(params.name);
+
+			var req = client.put(fileName, {
+				'Content-Length': params.length,
+				'Content-Type': mimeType,
+			});
+
+			params.readStream.pipe(req);
+
+			req.on('response', function(res){
+				if (res.statusCode !== 200) {
+					reject(new Error(res.statusCode));
+				} else {
+					var twentyFourHours = new Date(Date.now() + 24 * 60 * 60000);
+					var signedUrl = client.signedUrl(fileName, twentyFourHours);
+
+					// Resolve with an object following the "file" schema
+					resolve({
+						name: params.name,
+						url: signedUrl,
+						mime_type: mimeType,
+						expires: Math.round(twentyFourHours.getTime() / 1000)
+					});
+				}
+			});
+
+			req.on('error', reject);
+
+		});
+	};
+
+	/*
+		A multi-part variant of streamUpload; should only be used if there is
+		absolutely now way of uploading with the content-length header set.
+	*/
+	var streamMPUpload = function (params) {
+		return when.promise(function (resolve, reject) {
+
+			if (options.dev) {
+				// eslint-disable-next-line no-console
+				console.warn('Dev warning - avoid using streamMPUpload if possible.');
+			}
+
+			try { //Check if the lib has been included
+				require.resolve('knox-mpu-alt');
+			} catch (e) {
+				throw new Error('Please install `knox-mpu-alt` library and save it as a dependency to use streamMPUpload.');
+			}
+
+			var knoxMPU = require('knox-mpu-alt');
+
+			if (!_.isObject(options.aws)) {
+				return reject(new Error('AWS config not specified. Add a aws.json file in the connector folder, and pass it to falafel.'));
+			}
+
+			if (_.isUndefined(params.readStream) || _.isUndefined(params.readStream.pipe)) {
+				return reject(new Error('The object passed in must contain the property \'readStream\', referecing a read stream.'));
+			}
+
+			//Instantiate knox client
+			var client = knox.createClient({
+				key: options.aws.key,
+				secret: options.aws.secret,
+				bucket: 'workflow-file-uploads',
+				region: 'us-west-2'
+			});
+
+
+			// Generate a random file name and get mimeType
+			var fileName = guid(),
+				mimeType = params.contentType || mime.lookup(params.name);
+
+			//Instantiate knoxMPU instance with the knox client, stream, and metaData
+			new knoxMPU(
+				{
+					client: client,
+					objectName: fileName,
+					stream: params.readStream,
+					headers: {
+						'Content-Type': mimeType,
+					}
+				},
+				function(mpuUploadErr) {
+
+					if (mpuUploadErr) {
+						return reject(mpuUploadErr);
```

Comment:
maybe check if it's an Error

Reply:
The connector op will should handle that.